File: apps/backend/app/schemas/loyalty_metrics.py
from datetime import date
from typing import List
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class MetricSummary(BaseModel):
    template_id: UUID | None = None   # se None → todas
    total_cards: int
    unique_users: int
    total_stamps: int
    rewards_redeemed: int
    model_config = ConfigDict(from_attributes=True)


class SeriesPoint(BaseModel):
    day: date
    value: int


class ChartSeries(BaseModel):
    name: str          # “Cards emitted”, “Rewards redeemed”…
    points: List[SeriesPoint]


class MetricsCharts(BaseModel):
    template_id: UUID | None = None
    series: List[ChartSeries]

File: apps/portal-company/src/types/LoyaltyMetrics.ts
// src/types/LoyaltyMetrics.ts

export interface MetricSummary {
  template_id: string | null;
  total_cards: number;
  unique_users: number;
  total_stamps: number;
  rewards_redeemed: number;
}

export interface SeriesPoint {
  day: string;    // “YYYY-MM-DD”
  value: number;
}

export interface ChartSeries {
  name: string;
  points: SeriesPoint[];
}

export interface MetricsCharts {
  template_id: string | null;
  series: ChartSeries[];
}
